enterprise/internal/a8n/webhooks.go

```diff
@@ -364,6 +384,64 @@ func (*GitHubWebhook) pullRequestReviewCommentEvent(e *gh.PullRequestReviewComme
 	return &comment
 }
 
+func (h *BitbucketServerWebhook) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	e, err := h.parseEvent(r)
+	if err != nil {
+		respond(w, err.code, err)
+		return
+	}
+
+	pr, ev := h.convertEvent(e)
+	if pr == 0 || ev == nil {
+		respond(w, http.StatusOK, nil) // Nothing to do
+		return
+	}
+
+	if err := h.upsertChangesetEvent(r.Context(), pr, ev); err != nil {
+		respond(w, http.StatusInternalServerError, err)
+	}
+}
+
+func (h *BitbucketServerWebhook) parseEvent(r *http.Request) (interface{}, *httpError) {
+	args := repos.StoreListExternalServicesArgs{Kinds: []string{"BITBUCKETSERVER"}}
+	es, err := h.Repos.ListExternalServices(r.Context(), args)
+	if err != nil {
+		return nil, &httpError{http.StatusInternalServerError, err}
+	}
+
+	payload, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, &httpError{http.StatusInternalServerError, err}
+	}
+
+	var secrets [][]byte
+	for _, e := range es {
+		c, _ := e.Configuration()
+		hook, ok := c.(*schema.BitbucketServerConnection)
+		if !ok {
+			continue
+		}
+		secrets = append(secrets, []byte(hook.Webhooks.Secret))
+	}
+
+	sig := r.Header.Get("X-Hub-Signature")
+	for _, secret := range secrets {
+		if err = gh.ValidateSignature(sig, payload, secret); err == nil {
+			break
+		}
+	}
+
+	if len(secrets) == 0 || err != nil {
+		return nil, &httpError{http.StatusUnauthorized, err}
+	}
+
+	return nil, &httpError{http.StatusInternalServerError, fmt.Errorf("unimplemented")}
```

Comment:
Is this intentional as "unimplemented"? 
```suggestion
	return nil, &httpError{http.StatusInternalServerError, errors.New("unimplemented")}
```